src/utils.ts

```diff
@@ -166,3 +166,27 @@ export async function genEnvFile(buildDir: string) {
     // Ignore errors, Meson could be too old to support --dump-format.
   }
 }
+
+export async function patchCompileCommands(buildDir: string) {
+  // Remove ccache from compile commands because they confuse Intellisense:
+  // https://github.com/microsoft/vscode-cpptools/issues/7616
+  try {
+    var filePath = path.join(buildDir, "compile_commands.json")
+    var db = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+  } catch {
+    return;
+  }
+  var changed = false;
+  for (var i in db) {
+    // FIXME: This should use proper shlex.split() and shlex.join()
+    var cmd = db[i]['command'].split(" ");
+    if (cmd[0].endsWith("ccache")) {
+      cmd.shift();
+      db[i]["command"] = cmd.join(" ");
+      changed = true;
+    }
+  }
+  if (changed) {
+    fs.writeFileSync(filePath, JSON.stringify(db, null, "  "));
+  }
```

Comment:
you may be intesrsted in `for (let i of db)`, which is equivalent to python's `dict.values()`. Alternatively, I think this could be rewritten as:
```typescript
let changed = false;
Object.values(db).forEach((entry) => {
  let cmd = entry.command.split(" ");
  if (cmd[0].endsWith("ccache")) {
    cmd.shift();
    entry.command = cmd.join(" ");
    changed = true;
});
```

using these functional patterns is more idiomatic for javascript/typescript than writing for loops.

Also, in javascript/typescript `foo.bar` is the same as `foo["bar"]`. I haven't decided yet whether this is terrifying or amazing

Reply:
> Object.values(db).forEach((entry) => {

did that

> Also, in javascript/typescript foo.bar is the same as foo["bar"]

I think it needs to be propertly typed for that, at least the syntax highlighter is not happy with `entry.command`.